include/mbedtls/ssl_internal.h

```diff
@@ -420,6 +420,8 @@ struct mbedtls_ssl_handshake_params
                     const unsigned char *, size_t,
                     unsigned char *, size_t);
 
+    mbedtls_ssl_ciphersuite_t const *ciphersuite_info;
```

Comment:
Note (no change required, just thinking out loud it case it happens to spark something interesting): we already store the ciphersuite number as an `int` in `ssl_session`. So this field here is mainly useful as a cache for the resolution of this `int` to an `mbedtls_ssl_ciphersuite_t` pointer. I'm not sure what the effect on performance and/or code size would be of getting rid of that cache (which would save 4 bytes of RAM on 32-bit arches, and make the design a bit more orthogonal by eliminating redundancy).